ttnn/cpp/ttnn/operations/data_movement/transpose/transpose.cpp

```diff
@@ -10,37 +10,117 @@
 #include "ttnn/operations/data_movement/permute/permute.hpp"
 #include "ttnn/operations/data_movement/transpose/transpose.hpp"
 #include "ttnn/cpp/ttnn/operations/copy.hpp"
+#include "ttnn/cpp/ttnn/operations/data_movement/pad/pad.hpp"
+#include "ttnn/cpp/ttnn/operations/data_movement/slice/slice.hpp"
 
 
 namespace ttnn::operations::data_movement {
 
 namespace detail {
+uint32_t round_up(uint32_t value, uint32_t multiple) {
+    if (multiple == 0) {
+        return value;
+    } else if (value == 0) {
+        return multiple;
+    }
+    return (value % multiple == 0) ? value : ((value + multiple - 1) / multiple) * multiple;
+}
+
 inline Tensor transpose_(const Tensor &a, TransposeOpDim transpose_dim, const MemoryConfig& output_mem_config) {
     bool pad_c = false;
+    bool tiled_only = false;
     bool pad_n = false;
-
+    uint32_t ROW_MAJOR_STICK_WIDTH = 16; // this is a highly restrictive constraint on the RM transpose_wh kernel, and with all the other bugs/limitations we should rewrite it
```

Comment:
constexpr will probably satisfy the tool - should be applied anyways. 

General comment, not blocker for this PR - is this constrained based on noc alignment constraints or something else?  If so I'd recommend deriving based off that base level constraint and having that referenced from some common location (in some cases may have different values by arch).

Reply:
It's actually FACE_WIDTH I believe, will add that

It's a constraint on the OP/kernel for perf reasons (I'm guessing). Essentially it's at minimum one face (16 as a result) since the kernel is tilizing within the op as it loads into dest, transposes, then untilizes as its pushing them out. I imagine it was just a bit difficult to do this without aligned faces.

I will likely rework the logic to account for non-aligned dims.